source/MRViewer/MRRenderMeshObject.cpp

```diff
@@ -1020,6 +1119,49 @@ RenderBufferRef<uint8_t> RenderMeshObject::loadTexturePerFaceTextureBuffer_()
     return buffer;
 }
 
+RenderBufferRef<VertId> RenderMeshObject::loadPointValidIndicesBuffer_()
+{
+    auto& glBuffer = GLStaticHolder::getStaticGLBuffer();
+    if ( !dirtyPointPos_ || !objMesh_->mesh() )
+        return glBuffer.prepareBuffer<VertId>( pointValidSize_, !pointValidBuffer_.valid() );
+
+    MR_NAMED_TIMER( "mesh_points_dirty_valid_indices" );
+
+    const auto& topology = objMesh_->mesh()->topology;
+    const auto& validPoints = topology.getValidVerts();
+    pointValidSize_ = int( validPoints.count() );
+    auto buffer = glBuffer.prepareBuffer<VertId>( pointValidSize_ );
+    if ( cornerMode )
+    {
+        auto unprocessedPoints = validPoints;
+        auto numF = topology.lastValidFace() + 1;
+        for ( int f = 0, out = 0; f < numF; ++f )
+        {
+            if ( !topology.hasFace( FaceId( f ) ) )
+                continue;
+            const auto triVerts = topology.getTriVerts( FaceId( f ) );
+            for ( int i = 0; i < 3; ++i )
+            {
+                VertId v = triVerts[i];
+                if ( !unprocessedPoints.test( v ) )
+                    continue;
+                unprocessedPoints.set( v, false );
+                buffer[out++] = VertId( v );
+            }
+        }
```

Comment:
please see example in `loadVertUVBuffer_`

Reply:
reworked